### PR TITLE
feat: convert specs to TS

### PR DESCRIPTION
**Pull request by cabreraalex**
# Description

I think this makes it easier to maintain, create views, etc.

Also swapped the validation library to the more common and supported one.

### CODE REVIEW

File: src/lib/schema.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+export enum ViewTypeEnum {
```
Comment (Sparkier) [inline src/lib/schema.ts:1]: Generally find it a bit weird in TS to have `Enum` and `Type` in type/enum names. We don't do this typically in Hub. Maybe rather `ViewType`, `DisplayType`, `View`, `List`, ...

File: src/lib/views/schema.ts [hunk 2]
```diff
@@ -1,120 +0,0 @@
...
-			output: { $ref: '#/definitions/separatedValuesType' }
-		}
-	},
-	else: {
-		properties: {
-			data: allRefs,
-			label: allRefs,
-			output: allRefs
-		}
-	}
```
Comment (Sparkier) [inline src/lib/views/schema.ts:119]: This is not modeled in the new TS spec.

**Review by Sparkier — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -1,3 +1,29 @@
...
...Q9Ik0xNi40OTU2IDE3LjI0MzZMMjMuODUwNyAyNC41ODVMMTYuNDk1NiAzMS45NEwxNi40OTU2IDE3LjI0MzZaIiBmaWxsPSIjNjEyNTkzIiBmaWxsLW9wYWNpdHk9IjAuOCIvPgo8cGF0aCBkPSJNMTYuNTMyNiAxNi40Nzk1TDI0LjQ1MTUgMTYuNDc5NUwyNC40NTE1IDI0LjAyOEwxNi41MzI2IDE2LjQ3OTVaIiBmaWxsPSIjNjEyNTkzIiBmaWxsLW9wYWNpdHk9IjAuNiIvPgo8cGF0aCBkPSJNMjYuMTgxMyAxNi40MzI2TDI5LjA1MjUgMTYuNDMyNkwyOS4wNTI1IDE5LjM5NjRMMjYuMTgxMyAxNi40MzI2WiIgZmlsbD0iIzZBMUI5QSIgZmlsbC1vcGFjaXR5PSIwLjQiLz4KPHBhdGggZD0iTTI5LjU2MTkgMTYuNDMyNkwzMC43Nzg3IDE3LjY0OTRMMjkuNTUxNSAxOC44NzY2TDI5LjU2MTkgMTYuNDMyNloiIGZpbGw9IiM2QTFCOUEiIGZpbGwtb3BhY2l0eT0iMC4zIi8+CjxwYXRoIGQ9Ik0zMC4yNTY2IDE2LjQ3OTVMMzEuMTM2NSAxNi40Nzk1TDMxLjEzNjUgMTcuMzEzMUwzMC4yNTY2IDE2LjQ3OTVaIiBmaWxsPSIjNkExQjlBIiBmaWxsLW9wYWNpdHk9IjAuMiIvPgo8cGF0aCBkPSJNMzEuNDYwNiAxNi40Nzk1TDMxLjc4NDggMTYuODAzN0wzMS40NjA2IDE3LjEyNzlMMzEuNDYwNiAxNi40Nzk1WiIgZmlsbD0iIzYxMjU5MyIgZmlsbC1vcGFjaXR5PSIwLjEiLz4KPHBhdGggZD0iTTMyLjE4OCAxNi40NjkyTDMxLjk5MzIgMTYuNjY0MUwzMS43OTgzIDE2LjQ2OTJIMzIuMTg4WiIgZmlsbD0iIzYxMjU5MyIgZmlsbC1vcGFjaXR5PSIwLjEiLz4KPHBhdGggZD0iTTI1LjM0MjEgMTYuNDM4OUwyOC42NTMyIDE5Ljc1TDI1LjMyNDYgMjMuMDc4NkwyNS4zNDIxIDE2LjQzODlaIiBmaWxsPSIjNjEyNTkzIiBmaWxsLW9wYWNpdHk9IjAuNSIvPgo8L3N2Zz4K)](https://hub.zenoml.com)
```
[the remark this answers is not in the document]
Comment (cabreraalex) [reply README.md:8]: Same as other PR -> think we will use this generally for opening zeno projects in repos, this is just an example?

**Review by Sparkier — APPROVED** [review]
LGTM just need to make sure going forward that we test views on both table and non-table version.

**Pull request merged**